moonstone/filtering/mean_filtering.py

```diff
@@ -0,0 +1,120 @@
+
+    def filter(self):
+        self.steps.append('filtering_by_mean')
+        if not self.threshold:
+            self.compute_threshold_best_n_percent()
+        logger.info('Filtering with threshold set to %.2f...' % self.threshold)
+        filtered_df = self.df
+        filtered_df.drop(filtered_df[filtered_df.mean(axis=1) < self.threshold].index, inplace=True)
+        self.remaining_items_number = filtered_df.shape[0]
+        self.remaining_reads_number = filtered_df.sum().sum()
+        logger.info('Started with %i items and a total of %i reads' % (self.raw_items_number, self.raw_reads_number))
+        logger.info('Saving new matrix with %i items and %i reads' % (self.remaining_items_number,
+                                                                      self.remaining_reads_number))
+        logger.info('Retained %.2f%% of items and %.2f%% of reads' % (
+            self.remaining_items_number / self.raw_items_number * 100,
+            self.remaining_reads_number / self.raw_reads_number * 100)
+            )
+        return filtered_df
+
+    def plot_threshold_vs_remaining_data(self, write_html_file, items_name='items'):
+        '''  The x and y set below are are are either integers or floats.
+        Be aware that some operation will require an np.array  '''
+        x_items = list(self._items_dict.keys())  # get x values for plotting
+        x_reads = list(self._reads_dict.keys())
+        y_items = list(self._items_dict.values())  # get y values for plotting
+        y_reads = list(self._reads_dict.values())
+        '''  The figure visualizing the data and the filtering.  '''
+        fig = make_subplots(specs=[[{"secondary_y": True}]])  # To set a second y-axis.
+        fig.add_trace(
+            go.Scatter(x=x_items, y=y_items, name="Retained Items: left axis"),
+            secondary_y=False,
+        )
+        fig.add_trace(
+            go.Scatter(x=x_reads, y=y_reads, name="Retained Reads: right axis"),
+            secondary_y=True,
+        )
+        fig.add_trace(
+            go.Scatter(x=[self.threshold, self.threshold], y=[0, self.raw_reads_number], name="90% Reads Threshold"),
+            secondary_y=True,
+        )
+        # Add figure title
+        fig.update_layout(
+            title_text="Number of %s and reads in function of the threshold value" % items_name,
+            title_x=0.5
+        )
+        # Set x-axis title
+        fig.update_xaxes(title_text="threshold value")
+        # Set y-axes titles
+        fig.update_yaxes(title_text="number of %s" % items_name, secondary_y=False)
+        fig.update_yaxes(title_text="number of reads", secondary_y=True)
+        fig.show()
+
+        if write_html_file:
+            plotly.io.write_html(fig, write_html_file)
+
+    def visualize(self, write_html_file=False):
+        print(write_html_file)
+        try:
+            self._items_dict
+        except AttributeError:
+            FS_instance = FilteringStats(self.df)
```

Comment:
Variables names should be in lower case (check this about naming convention for more information: https://www.python.org/dev/peps/pep-0008/#naming-conventions).